src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD112AvoidNullReturnInNonAsyncTaskMethodAnalyzerTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿namespace Microsoft.VisualStudio.Threading.Analyzers.Tests
+{
+    using System.Threading.Tasks;
+    using Xunit;
+    using Verify = CSharpCodeFixVerifier<VSTHRD112AvoidNullReturnInNonAsyncTaskMethodAnalyzer, CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+    public class VSTHRD112AvoidNullReturnInNonAsyncTaskMethodAnalyzerTests
+    {
+        [Fact]
+        public async Task TaskOfTReturnsNull_Diagnostic()
+        {
+            var test = @"
+using System.Threading.Tasks;
+
+class Test
+{
+    public Task<object> GetTaskObj()
+    {
+        return null;
+    }
+}
+";
+            await new Verify.Test
+            {
+                TestCode = test,
+                ExpectedDiagnostics = { Verify.Diagnostic().WithSpan(8, 9, 8, 21), },
```

Comment:
Can we use the `[| |]` syntax to mark where the diagnostic is expected instead? It's easier to read and maintain.